dymos/load_case.py

```diff
@@ -51,83 +78,101 @@ def load_case(problem, previous_solution):
 
+    traj_paths = find_trajectories(problem.model)
     phase_paths = find_phases(problem.model)
 
-    if not phase_paths:
+    if not phase_paths and not traj_paths:
         return
 
-    prev_outputs = {v['prom_name']: {'value': v['value'], 'units': v['units']} for k, v in previous_solution['outputs']}
+    prev_vars = {}
+    prev_vars.update({v['prom_name']: {'value': v['value'], 'units': v['units']} for k, v in previous_solution['inputs']})
+    prev_vars.update({v['prom_name']: {'value': v['value'], 'units': v['units']} for k, v in previous_solution['outputs']})
 
     problem.final_setup()  # make sure list_inputs and list_outputs can work
 
     phase_io = {'inputs': problem.model.list_inputs(out_stream=None, units=True, prom_name=True),
                 'outputs': problem.model.list_outputs(out_stream=None, units=True, prom_name=True)}
 
-    phase_inputs = {v['prom_name']: {'value': v['value'], 'units': v['units']} for k, v in phase_io['inputs']}
-    phase_outputs = {v['prom_name']: {'value': v['value'], 'units': v['units']} for k, v in phase_io['outputs']}
+    phase_vars = {}
+    phase_vars.update({v['prom_name']: {'value': v['value'], 'units': v['units']} for k, v in phase_io['inputs']})
+    phase_vars.update({v['prom_name']: {'value': v['value'], 'units': v['units']} for k, v in phase_io['outputs']})
+
+    for traj_abs_path, traj in traj_paths.items():
+        traj_name = traj_abs_path.split('.')[-1]
+        for param_name, options in traj.parameter_options.items():
+            prev_match = [s for s in prev_vars if s.endswith(f'{traj_name}.parameters:{param_name}')]
+            if prev_match:
+                # In previous outputs
+                prev_data = prev_vars[prev_match[0]]
+                prev_val = prev_data['value']
+                prev_units = prev_data['units']
+            else:
+                raise Warning(f'Unable to find a value for {traj_name}.parameters:{param_name} in the restart file.')
+            problem.set_val(f'{traj.pathname}.parameters:{param_name}', prev_val[0, ...], units=prev_units)
 
     for phase_abs_path, phase in phase_paths.items():
         phase_name = phase_abs_path.split('.')[-1]
 
         # Get the initial time and duration from the previous result and set them into the new phase.
-        t_path = [s for s in phase_outputs if s.endswith(f'{phase_name}.timeseries.time_phase')][0]
+        prev_time_path = [s for s in prev_vars if s.endswith(f'{phase_name}.timeseries.time')][0]
 
-        t_initial = prev_outputs[t_path]['value'][0]
-        t_initial_units = prev_outputs[t_path]['units']
+        prev_time_val = prev_vars[prev_time_path]['value']
+        prev_time_units = prev_vars[prev_time_path]['units']
 
-        t_duration = prev_outputs[t_path]['value'][-1] - prev_outputs[t_path]['value'][0]
-        t_duration_units = t_initial_units
+        t_initial = prev_time_val[0]
```

Comment:
Kind of curious why we don't grab t_initial and t_duration from the time component inputs instead of computing it from the outputs. That seems more direct (and safer if we are loading some failed case where some comps didn't run because of a raised exception.)

Reply:
We need the entirety of the time vector to interpolate the state and control values (when the grid is different, so it just seemed easier to grab them there)

If you have a case that aborted due to an exception you probably shouldn't be able to successfully load it anyway.